mti_nma/steps/singlecell/singlecell_utils.py

```diff
@@ -0,0 +1,127 @@
+import numpy as np
+import pandas as pd
+import labkey as lk
+from lkaccess import LabKey as lka
+from skimage import transform as sktrans
+
+def query_data_from_labkey(CellLineId):
+
+    # Query for labkey data
+
+    filters = [
+            # 100X FOVs
+            lk.query.QueryFilter('Objective', '100.0', 'eq'),
+            # Production data
+            lk.query.QueryFilter('WellId/PlateId/PlateTypeId/Name', 'Production', 'contains'),
+            # Pipeline 4
+            lk.query.QueryFilter('WellId/PlateId/Workflow/Name', '4', 'contains'),
+            # Not celigo images
+            lk.query.QueryFilter('InstrumentId/Name', 'Celigo', 'neqornull'),
+            # Passed QC
+            lk.query.QueryFilter('QCStatusId/Name', 'Passed', 'eq'),
+            # Cell line (13 = Lamin)
+            lk.query.QueryFilter('SourceImageFileId/CellLineId/Name', CellLineId, 'contains')
+    ]
+
+    query_raw = lka(host="aics").select_rows_as_list(
+
+        schema_name = "microscopy",
+        query_name="FOV",
+        filter_array = filters,
+        columns = [
+            'FOVId',
+            'PixelScaleX',
+            'PixelScaleY',
+            'PixelScaleZ',
+            'SourceImageFileId/LocalFilePath'
+        ]
+
+    )
+
+    query_seg = lka(host="aics").select_rows_as_list(               
+
+        schema_name = 'fms',
+        query_name = 'FileUI',
+        filter_array = filters + [
+            # Nuclei segmentation only
+            lk.query.QueryFilter('ReadPath', 'nucWholeIndexImageScale', 'contains')
+        ],
+        columns = [
+            'FOVId',
+            'ReadPath',
+        ]
+
+    )
+
+    df_seg = pd.DataFrame(query_seg) # FOVIds are returned as list
+    df_seg = df_seg.rename(columns={'FOVId': 'FOVIdList', 'ReadPath': 'ReadPathSeg'})
+
+    df_raw = pd.DataFrame(query_raw)
+    df_raw = df_raw.rename(columns={'SourceImageFileId/LocalFilePath': 'ReadPathRaw'})
+
+    # Merging raw and seg tables
+
+    for index in df_seg.index:
+        fov_id = df_seg.FOVIdList[index]
+        if len(fov_id):
```

Comment:
Sorry - I don't quite understand what this if/block is doing

Reply:
Sorry. This should have been better explained. The problem here is that some of the FOVs on labkey do not have segmentation and therefore the list FOVIdList is empty. I added a comment to address that.